src/herbpy/action/rogue.py

```diff
@@ -0,0 +1,354 @@
+    @param render Render tsr samples during planning
+    """
+    if manip is None:
+        manip = robot.right_arm
+
+    if manip.GetName() != 'right':
+        raise prpy.exceptions.PrPyException('Pointing is only defined \
+                on the right arm.')
+
+    focus_trans = numpy.eye(4)
+    focus_trans[0:3, 3] = coord
+    point_tsr = robot.tsrlibrary(None, 'point', focus_trans, manip)
+
+    p = openravepy.KinBody.SaveParameters
+    with robot.CreateRobotStateSaver(p.ActiveManipulator):
+        robot.SetActiveManipulator(manip)
+        with prpy.viz.RenderTSRList(point_tsr, robot.GetEnv(), render=render):
+            robot.PlanToTSR(point_tsr, execute=True)
+    manip.hand.MoveHand(f1=2.4, f2=0.8, f3=2.4, spread=3.14)
+
+def Present(robot, coord, manip=None, render=True):
+    """
+    @param robot The robot performing the presentation
+    @param focus The 3-D coordinate in space or object that 
+                 is being presented
+    @param manip The manipulator to perform the presentation with. 
+                 This must be the right arm.
+    @param render Render tsr samples during planning
+    """
+    focus_trans = numpy.eye(4)
+    focus_trans[0:3, 3] = coord
+
+    present_tsr = robot.tsrlibrary(None, 'present', focus_trans, manip)
+
+    if manip is None:
+        manip = robot.right_arm
+
+    if manip.GetName() != 'right':
+        raise prpy.exceptions.PrPyException('Presenting is only defined \
+                on the right arm.')
+    
+    p = openravepy.KinBody.SaveParameters
+    with robot.CreateRobotStateSaver(p.ActiveManipulator):
+        robot.SetActiveManipulator(manip)
+        with prpy.viz.RenderTSRList(present_tsr, robot.GetEnv(), render=render):
+            robot.PlanToTSR(present_tsr, execute=True)
+    manip.hand.MoveHand(f1=1, f2=1, f3=1, spread=3.14)
+
+
+def Sweep(robot, start_coords, end_coords, manip=None, margin=0.3, render=True):
+    """
+    @param robot The robot performing the sweep
+    @param start The object or 3-d position that marks the start
+    @param end The object of 3-d position that marks the end
+    @param manip The manipulator to perform the sweep
+    @param margin The distance between the start object and the hand,
+                  so the vertical space between the hand and objects. 
+                  This must be enough to clear the objects themselves.
+    @param render Render tsr samples during planning
+    """
+    if manip is None:
+        manip = robot.GetActiveManipulator()
+
+    #ee_offset : such that the hand, not wrist, is above the object
+    #hand_pose : places the hand above the start location
+    if manip.GetName() == 'right':
+        hand = robot.right_hand
+        ee_offset = -0.2
+        hand_pose = numpy.array([[ 0, -1, 0, start_coords[0]],
+                                 [ 0,  0, 1, (start_coords[1]+ee_offset)],
+                                 [-1,  0, 0, (start_coords[2]+margin)],
+                                 [ 0,  0, 0, 1]])
```

Comment:
You should either explicitly pass `dtype='float'` or insure that at least one literal value in the array is a floating point number (e.g. replace `0` with `0.`). Otherwise, this could create an `int` array, which can lead to all sorts of unpleasant surprises.

This applies to all of the hard-coded transformations.